kyo-playwright/shared/src/main/scala/kyo/Browser.scala

```diff
@@ -0,0 +1,962 @@
+                            .setContentType(contentType))
+                )
+            )
+
+        case class Unroute(url: String)
+            extends Op(_.unroute(url))
+
+        case object UnrouteAll
+            extends Op(_.unrouteAll())
+
+        // Forms
+
+        case class Fill(selector: String, text: String)
+            extends Op(_.fill(selector, text))
+
+        case class Press(selector: String, key: String)
+            extends Op(_.press(selector, key))
+
+        case class Check(selector: String)
+            extends Op(_.check(selector))
+
+        case class Uncheck(selector: String)
+            extends Op(_.uncheck(selector))
+
+        case class Select(selector: String, value: String)
+            extends Op(_.selectOption(selector, value).asScala.toList)
+
+        case class SelectMultiple(selector: String, values: List[String])
+            extends Op(_.selectOption(selector, values.toArray).asScala.toList)
+
+        // DOM
+
+        case class InnerText(selector: String)
+            extends Op(_.innerText(selector))
+
+        case class InnerHTML(selector: String)
+            extends Op(_.innerHTML(selector))
+
+        case class TextContent(selector: String)
+            extends Op(p => Maybe(p.textContent(selector)))
+
+        case class Attribute(selector: String, name: String)
+            extends Op(p => Maybe(p.getAttribute(selector, name)))
+
+        case class FindByText(text: String)
+            extends Op(_.locator(s"text='${escapeString(text)}'").first().toString())
+
+        case class FindByRole(role: String, name: String)
+            extends Op(page =>
+                val nameOption = if name.isEmpty then "" else s"name='${escapeString(name)}'"
+                val selector = if nameOption.isEmpty then
+                    s"role=${role}"
+                else
+                    s"role=${role}[${nameOption}]"
+                page.locator(selector).first().toString()
+            )
+
+        case class GetElementPosition(selector: String)
+            extends Op(page =>
+                val bounds = page.locator(selector).boundingBox()
+                ElementPosition(
+                    bounds.x,
+                    bounds.y,
+                    bounds.width,
+                    bounds.height,
+                    bounds.x + bounds.width / 2,
+                    bounds.y + bounds.height / 2
+                )
+            )
+
+        case class Exists(selector: String)
+            extends Op(page =>
+                try
+                    page.waitForSelector(selector) != null
+                catch
+                    case _: PlaywrightException => false
+            )
+
+        case class Count(selector: String)
+            extends Op(page => page.locator(selector).count())
+
+        // Content
+
+        case object GetStatus
+            extends Op(page => Status(page.url(), page.title()))
+
+        case object ReadableContent
```

Comment:
This op uses two JS libraries to extract the readable content and then transform it into markdown.

Reply:
Yes, but it seems the only option. The injection JS disappears once there's a page refresh or navigation. I haven't noticed issues with multiple readablecontent calls as well